tests/kafkatest/tests/streams/streams_broker_down_resilience_test.py

```diff
@@ -112,3 +127,89 @@ def test_streams_resilient_to_broker_down(self):
         self.assert_produce_consume("after_broker_stop")
 
         self.kafka.stop()
+
+    def test_streams_runs_with_broker_down_initially(self):
+        self.kafka.start()
+        node = self.kafka.leader(self.inputTopic)
+        self.kafka.stop_node(node)
+
+        configs = self.get_configs(extra_configs=",application.id=starting_wo_broker_id")
+
+        # start streams with broker down initially
+        processor = StreamsBrokerDownResilienceService(self.test_context, self.kafka, configs)
+        processor.start()
+
+        processor_2 = StreamsBrokerDownResilienceService(self.test_context, self.kafka, configs)
+        processor_2.start()
+
+        processor_3 = StreamsBrokerDownResilienceService(self.test_context, self.kafka, configs)
+        processor_3.start()
+
+        broker_unavailable_message = "Broker may not be available"
+
+        # verify streams instances unable to connect to broker, kept trying
+        self.wait_for_verification(processor, broker_unavailable_message, processor.LOG_FILE, 100)
+        self.wait_for_verification(processor_2, broker_unavailable_message, processor_2.LOG_FILE, 100)
+        self.wait_for_verification(processor_3, broker_unavailable_message, processor_3.LOG_FILE, 100)
+
+        # now start broker
+        self.kafka.start_node(node)
+
+        # assert streams can process when starting with broker down
+        self.assert_produce_consume("running_with_broker_down_initially", num_messages=9)
+
+        message = "processed3messages"
+        # need to show all 3 instances processed messages
+        self.wait_for_verification(processor, message, processor.STDOUT_FILE)
+        self.wait_for_verification(processor_2, message, processor_2.STDOUT_FILE)
+        self.wait_for_verification(processor_3, message, processor_3.STDOUT_FILE)
+
+        self.kafka.stop()
+
+    def test_streams_should_scale_in_while_brokers_down(self):
+        self.kafka.start()
+
+        configs = self.get_configs(extra_configs=",application.id=shutdown_with_broker_down")
+
+        processor = StreamsBrokerDownResilienceService(self.test_context, self.kafka, configs)
+        processor.start()
+
+        processor_2 = StreamsBrokerDownResilienceService(self.test_context, self.kafka, configs)
+        processor_2.start()
+
+        processor_3 = StreamsBrokerDownResilienceService(self.test_context, self.kafka, configs)
+        processor_3.start()
+
+        # need to wait for rebalance  once
+        with processor_3.node.account.monitor_log(processor_3.LOG_FILE) as monitor:
```

Comment:
There is a race condition to start processor before the `with` clause, as it may print some console outputs that would not be captured in the `with .. until` verification. See https://github.com/apache/kafka/pull/4348/files as a reference.

Reply:
ack, I'll change to use the `ssh_output` option, I prefer that approach anyway.